google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -120,6 +121,15 @@ public CancellableRunnable createDeployRunner(
       throw new RuntimeException("Invalid deployment source selected for deployment");
     }
 
+    if (CloudSdkService.getInstance().validateCloudSdk().contains(
+        CloudSdkValidationResult.CLOUD_SDK_NOT_FOUND)) {
+      callback.errorOccurred(GctBundle.message("appengine.cloudsdk.location.invalid.message") + " "
+          + CloudSdkService.getInstance().getSdkHomePath());
+      logger.warn(GctBundle.message("appengine.cloudsdk.location.invalid.message") + " "
```

Comment:
why does the first check throw an exception, but this one log a warning? Also you're still using the resource bundle in your log message.